File: StudentEnrolmentApp/StudentEnrolmentApplication/StudentEnrolment.Core/Models/StudentDetailsModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace StudentEnrolment.Core.Models
{
    public class StudentDetailsModel
    {
        public int StudentId { get; set; }
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string KnownAs { get; set; } = string.Empty;
        public string DisplayName { get; set; } = string.Empty;
        public string DateOfBirth { get; set; } = string.Empty;
        public string Gender { get; set; } = string.Empty;
        public string UniversityEmail { get; set; } = string.Empty;
        public string NetworkId { get; set; } = string.Empty;
        public string HomeOrOverseas { get; set; } = string.Empty;
        public List<EnrolmentDetailsModel>? CourseEnrolment { get; set; }
    }
}

File: StudentEnrolmentApp/StudentEnrolmentUI/src/app/models/student-details.model.ts
import { Course } from "./course-model";

export class StudentDetails {
  studentId: number = 0;
  firstName = "";
  lastName = "";
  knownAs = "";
  displayName = "";
  dateOfBirth = "";
  gender = "";
  universityEmail = "";
  networkId = "";
  homeOrOverseas = "";
  courseEnrolment: CourseEnrolment[] = [];
}

export class CourseEnrolment {
  enrolmentId="";
  academicYear = "";
  yearOfStudy = "";
  occurrence = "";
  modeOfAttendance = "";
  enrolmentStatus = "";
  courseEntryDate = "";
  expectedEndDate = "";
  course: Course = new Course;
}
